components/proxy/src/main/java/com/hotels/styx/ServerConfigSchema.java

```diff
@@ -70,7 +70,7 @@
 
         STYX_SERVER_CONFIGURATION_SCHEMA_BUILDER = newDocument()
                     .rootSchema(object(
-                            field("proxy", object(
+                            optional("proxy", object(
```

Comment:
I think we need to move at least the compressResponses attribute somewhere else if it is optional now. 
Don't we need any other of the attributes? factories, bossThreadsCount ... It seems bossThreadsCount was always  applied to each listener/ServerBooStrap individually (we aren't sharing the NioEventLoopGroup between different BootStraps).

Reply:
1. `compressResponses` attribute, and the other missing attributes would still work with the old `proxy` attribute. I will add them as a separate increment to the server object. I can do it straight after the PR is merged.

2. Regarding The executors & thread counts. This is a know limitation and I think mentioned in the PR description. We need to think about a specific mechanism of configuring and injecting event loops into various kinks of styx objects: servers, routing objects, and providers. I'm thinking to introduce a separate `executors` block where you declare named executors, which can be then called into action from the other styx objects. For example:

```
executors:
  ioClient:
    type: NettyEventloop
    config:
       threads: 4
       channelType: client

  ioServer:
    type: NettyEventloop
    config:
      thread: 3
      channelType: server

servers:
  myServer:
    type: HttpServer
    config:
      port: 8080
      handler: hostProxy
      eventLoop: ioServer

routintObjects:
  hostProxy:
    type: HostProxy
    config:
       host: abc:8080
       eventLoop: ioClient

```

WDYT?

Reply:
Sounds good to me.

Reply:
Raised an issue for missing content compression: https://github.com/HotelsDotCom/styx/issues/593